sdk/src/Services/S3/Custom/Transfer/Internal/_async/MultipartUploadCommand.async.cs

```diff
@@ -240,22 +238,16 @@ private async Task UploadNonSeekableStreamAsync(TransferUtilityUploadRequest req
                                 bool isLastPart = readBytesCount == 0;
                                 var partSize = nextUploadBuffer.Position;
                                 nextUploadBuffer.Position = 0;
-                                var partResponse = await _s3Client.UploadPartAsync(new UploadPartRequest()
-                                {
-                                    BucketName = request.BucketName,
-                                    Key = request.Key,
-                                    UploadId = initiateResponse.UploadId,
-                                    InputStream = nextUploadBuffer,
-                                    PartSize = partSize,
-                                    PartNumber = partNumber,
-                                    IsLastPart = isLastPart
-                                }).ConfigureAwait(false);
+                                UploadPartRequest uploadPartRequest = ConstructUploadPartRequestForNonSeekableStream(nextUploadBuffer, partNumber, partSize, isLastPart, initiateResponse);
+
+                                var partResponse = await _s3Client.UploadPartAsync(uploadPartRequest).ConfigureAwait(false);
                                 Logger.DebugFormat($"Uploaded part {partNumber}. (Last part = {isLastPart}, Part size = {partSize}, Upload Id: {initiateResponse.UploadId}");
```

Comment:
Can you fix the `DebugFormat` calls to not use string interpolation using the `$`. This is mean to pass in a format string with parameters so that we don't make allocations creating a debug string that is never used. It is done a few places in  this function and possible other places updated for Unseekable streams.

```
Logger.DebugFormat($"Uploaded part {0}. (Last part = {1}, Part size = {2}, Upload Id: {3}", partNumber, isLastPart, partSize, initiateResponse.UploadId);
```

Reply:
Fixed. Please review.